poollet/machinepoollet/controllers/machine_controller_networkinterface.go

```diff
@@ -245,22 +247,33 @@ func (r *MachineReconciler) prepareIRINetworkInterface(
 		r.Eventf(machine, corev1.EventTypeNormal, events.NetworkInterfaceNotReady, "Network interface %s network %s not found", nic.Name, networkKey.Name)
 		return nil, false, nil
 	}
-	labels, err := r.iriNetworkInterfaceLabels(nic)
+	nicLabels, err := r.iriNetworkInterfaceLabels(nic)
 	if err != nil {
 		return nil, false, fmt.Errorf("error preparing iri networkinterface labels: %w", err)
 	}
 
+	networkLabels, err := r.iriNetworkLabels(network)
+	if err != nil {
+		return nil, false, fmt.Errorf("error preparing iri network labels: %w", err)
+	}
+
+	attributes := make(map[string]string)
+	for k, v := range nic.Spec.Attributes {
+		attributes[k] = v
+	}
+
+	attributes[v1alpha1.NICLabelsAttributeKey] = string(maps.MustMarshalJSON(nicLabels))
+	attributes[v1alpha1.NetworkLabelsAttributeKey] = string(maps.MustMarshalJSON(networkLabels))
+
```

Comment:
Can't we avoid all these checks if we update the `broker` to don't report what labels are set to the object?